server/service/schedule/schedule.go

```diff
@@ -199,6 +203,27 @@ func (s *Schedule) Start() {
 						ctxerr.Handle(s.ctx, err)
 					}
 				}
+
+				totalRunningTime := time.Since(newStart)
+				schedInterval = s.getSchedInterval()
+
+				if totalRunningTime > schedInterval {
+					level.Info(s.logger).Log("cron", s.name, "msg", "total runtime (%d) exceeded schedule interval (%d)")
+				}
+
+				// set new wait to the remaining duration after dividing the total running time by the schedule interval
+				newWait = schedInterval - (totalRunningTime % schedInterval)
```

Comment:
Hm, I think instead of `newWait`, I would call this `remainingWait`. Because the idea is that if the schedInterval is 1hr and totalRunningTime is 1.5 hrs then we only wait 0.5 hrs until the next time we run this so it continues to run once an hour.

If it takes over 2*schedInterval, then we'll just wait schedInterval.

Do I understand this correctly? If so, could we expand the comment here so that these cases are clear? It might've been just me, but it took me a bit to wrap my head around this.

Reply:
> If it takes over 2*schedInterval, then we'll just wait schedInterval.

We'd always wait the remainder. We're probably saying the same thing here but here's a concrete example just to be explicit. Using the example you gave, if the job took 2.1 hours to run, we'd wait .9 hours for the next run.